### ISSUE TITLE
Configuring and starting a Wash cycle

### ISSUE THREAD

**Issue by ArthFink** [home-assistant-HomeWhiz/home-assistant-HomeWhiz#211]
I am not sure if I just overlooked something, but I would like to start a washing cycle directly though home assistant and maybe also be able to set the finish time.

**Comment by rubengonzalez96** [comment]
Yes, I would like something like this too. I tried to make an script with home assistant visual editor but I cannot not getting anything else than a "beep" on my washing machine with no effect. This means I'm on the right way but some help would be fine!
I tried to set on my machine, synthetic program with 40ºC and 1200rpm but it doesn't make any change on the washing machine panel (deleted ids for security)

`alias: alias
sequence:
  - device_id: ...
    domain: select
    entity_id:  ...
    type: select_option
    option: device_state_on
  - device_id:  ...
    domain: select
    entity_id:  ...
    type: select_option
    option: program_synthetics
  - device_id:  ...
    domain: select
    entity_id:  ...
    type: select_option
    option: temperature_40
  - device_id:  ...
    domain: select
    entity_id:  ...
    type: select_option
    option: 1200rpm
icon: mdi:washing-machine
mode: single
`

**Comment by TechHummel** [comment]
Are you able to start the washing cycle or set a finish time in the mobile app?

**Comment by ArthFink** [comment]
Yes it's possible. You need to select the remote control mode, it's basically like starting a program from the Maschine but without selecting all options. Then in the app you can basically choose as if you would use the dial or even tell what's kind of clothes are in the drum and it gives you some options for programs. 

But even just starting a program would be a great addition to the integration. The other features are not really needed.

tldr: yes a core feature of the app.

**Comment by UZi02** [comment]
I would appreciate this feature, too. I'd use it for starting the washing machine once the solar panels begin feeding power into the grid.

In the app you can set the program with all the options you already have in there (switch and select options). Maybe the select.waschmaschine_zusatnd (this is how it is called in my configuration) would do the job already?

**Comment by guinius** [comment]
I really want this for my dishwasher as well, and in fact I noticed with the dishwasher when I set a program on the app with time delay, some of the controls change in HA. In the screenshot, you can see “State” is set to “time delay active” but what I have noticed is that if I turn on and place the dishwasher into app mode, but then try to change these control settings on in HA instead of opening the app, nothing happens.

![A564FE1D-5EB7-4DE1-A30E-536C7B00162C](https://github.com/home-assistant-HomeWhiz/home-assistant-HomeWhiz/assets/152700423/a5d7bbb4-21f8-405d-94ec-b1d3e16e8590)
![9E273797-F25D-4AD4-8D98-05FBCF650DDC](https://github.com/home-assistant-HomeWhiz/home-assistant-HomeWhiz/assets/152700423/52bf08e3-f01a-4233-91c7-1ccaf799a177)


Any suggestions welcome!

**Comment by UZi02** [comment]
So I tried with my Beko laundry. When I set it in the remote control state I can set the program etc like pet hair removal (Tierhaarentfernung) on the controls. And when I set the state (Zustand) to "Running" (Läuft) it starts immediately.

**Comment by guinius** [comment]
> So I tried with my Beko laundry. When I set it in the remote control state I can set the program etc like pet hair removal (Tierhaarentfernung) on the controls. And when I set the state (Zustand) to "Running" (Läuft) it starts immediately. But setting the laundry first into the remote control sate is essential.

This does work you are right, but only useful to start the cycle at that time, I was hopping to be able to use the delay function on the dishwasher. I can select the state to Time Delay Active, but there is no way of setting the time delay. I tried building an automation to set and Start the cycle, the pause it. Then at a set time (when my electricity is cheaper) start it, but when paused for more than 3min the dishwasher turns off and Wi-Fi connection is lost.

any other suggestions are welcome! Is there a way to set the time delay?

**Comment by izotop510** [comment]
I have a beko washing machine with bluetooth connection, I use this workaround and it works for me, I use an script that configure the wash program settings, then the script sets the state in running mode,  waits 3 seconds and the script changes the state to pause mode, at this moment I can wait all the time I want to set the washing machine state in running mode and the selected program will continue his process ( I did this last step changing the state manually and automatically with an script and both ways work).

**Comment by UZi02** [comment]
You're right. On my Beko laundry I even can't set a delay time in the controls. Just see in the sensors if one is set. Although in the Homewhiz app I can't set a delay time just a time when it should be finished.

**Comment by TechHummel** [comment]
I'll close this, as you seem to all have found a solution 😊

**Comment by tszyszko** [comment]
Anyone have any workaround for keeping the dishwasher switched on? It just turns off if left in pause mode
